taxonomy-editor-frontend/src/pages/editentry/ListTranslations.jsx

```diff
@@ -1,94 +1,353 @@
+            const duplicateMainLang_renderedTranslations = mainLang_renderedTranslations.filter(obj => !(index === obj.index))
+            setMainLang_renderedTranslations(duplicateMainLang_renderedTranslations); // Set state
+
+            // Updated tags assigned for later use
+            tagsToBeInserted = duplicateMainLang_renderedTranslations.map(el => (el.tag))
+        }
+        // State of "renderedTranslations" is updated according to format used
+        else {
+            let newRenderedTranslations = []
+            renderedTranslations.forEach((allTagsObj) => {
+                if (allTagsObj['languageCode'] === key) {
+                    const unDeletedTags = allTagsObj['tags'].filter((tagObj) => !(tagObj.index === index));
+                    newRenderedTranslations.push({
+                        'languageCode' : key,
+                        'tags' : unDeletedTags
+                    })
+                    // Updated tags assigned for later use
+                    tagsToBeInserted = [...unDeletedTags].map(el => (el.tag));
+                }
+                else {
+                    newRenderedTranslations.push(allTagsObj)
+                }
+            })
+            setRenderedTranslations(newRenderedTranslations) // Set state
+        }
+        // Set state of main NodeObject
+        setNodeObject(prevState => {
+            const newNodeObject = {...prevState};
+            newNodeObject['tags_'+key] = tagsToBeInserted;
+            return newNodeObject
+        })
     }
 
     return ( 
-        <Box sx={{ml: 4}} className="translations">
+        <Box sx={{ml: 4}}>
             {/* Title */}
-            <Typography sx={{mt: 4, mb: 1}} variant='h5'>Translations</Typography>
+            <Stack direction="row" alignItems="center">
+                <Typography sx={{mt: 4, mb: 1}} variant='h5'>Translations</Typography>
+                <IconButton sx={{mt: 3.5, ml: 1, color: "#808080"}} onClick={handleOpen}>
+                    <AddBoxIcon />
+                </IconButton>
+            </Stack>
+
             {/* Main Language */}
-            <Typography variant='h6'>
-                { ISO6391.getName(nodeObject.main_language) }
-            </Typography>
+            <Stack direction="row" alignItems="center">
+                <Typography variant='h6'>
+                    { nodeObject && ISO6391.getName(nodeObject.main_language) }
+                </Typography>
+                <IconButton sx={{ml: 1, color: "#808080"}} onClick={(e) => handleAdd(nodeObject.main_language, e)}>
+                    <AddBoxIcon />
+                </IconButton>
+            </Stack>
+            
             {/* Render main language tags */}
             <Typography variant='h6'> 
-                {
-                    nodeObject["tags_"+nodeObject['main_language']].map((tag, index) => {
+                { nodeObject && 
+                    mainLang_renderedTranslations.map((tagObj) => {
```

Comment:
nit: `mainLang_renderedTranslations.map((tagObj) => {` could be `mainLang_renderedTranslations.map(({index, tag}) => {`